### PR TITLE
add support for webpack 4

### LINKED ISSUE #83: Webpack 4 support
Hi, I am thinking to upgrade to Webpack 4. 

Just want to ask if it is in your plans to add its support?

Cheers

### PR DESCRIPTION
**Pull request by gseguin**
Fixes #83

### CODE REVIEW

File: test/ProductionModePlugin.js [hunk 1]
```diff
@@ -33,26 +46,26 @@ const mkWebpackConfig = (options) => ({
...
-        return module.request && module.request.startsWith(nodeModules);
-      }
-    }),
-    new webpack.optimize.CommonsChunkPlugin({
-      name: "runtime",
-      filename: "runtime.js",
-      minChunks: Infinity
-    })
+    )
+    // new webpack.optimize.CommonsChunkPlugin({
```
Comment (gseguin) [inline test/ProductionModePlugin.js]: Oh oops, gotta remove those

File: test/ProductionModePlugin.js [hunk 2]
```diff
@@ -17,8 +17,21 @@ const mkWebpackConfig = (options) => ({
...
-    path: options.outputPath,
-    filename: "app.js"
+    path: options.outputPath
+  },
+  optimization: {
+    minimize: false,
+    runtimeChunk: "single",
+    splitChunks: {
+      chunks: "all",
+      cacheGroups: {
```
Comment (gseguin) [inline test/ProductionModePlugin.js]: This might actually be not needed

Comment (gseguin) [reply test/ProductionModePlugin.js]: Verified, it is

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "globalize-webpack-plugin",
-  "version": "2.1.1",
+  "version": "3.0.0",
```
Comment (mimckenna) [inline package.json]: You might want to update the [README.md](https://github.com/rxaviers/globalize-webpack-plugin/blob/master/README.md) to document changes for Webpack4.

**Comment by rxaviers** [comment]
Thanks @gseguin! It's good to see you around!

I'm merging your PR and (in a separate commit) I am going to drop webpack 3 support because it simplifies code readability and makes testing easier (otherwise we'd have to test both webpack versions). All without much impact (who needs webpack 3 can use 2.x). I am going to publish webpack 4 support in 3.x version.

**Comment by rxaviers** [comment]
Published pre-release 4.0.0-0.

File: package.json (hunk 3 above)
Comment (rxaviers) [reply package.json]: Yeap Mike. I went ahead and updated documentation accordingly.

**Comment by rxaviers** [comment]
Edit: Published pre-release 3.0.0-0.

**Pull request merged**